spark/src/main/scala/org/apache/spark/sql/delta/TableFeature.scala

```diff
@@ -284,28 +282,37 @@ object TableFeature {
     allSupportedFeaturesMap.get(featureName.toLowerCase(Locale.ROOT))
 
   /**
-   * Extracts the downgraded feature name from protocol downgrade commit info.
+   * Extracts the removed feature name by comparing current and base protocols.
+   * Returns None if there is no removed feature.
```

Comment:
This function's signature is very specific to the current implementation where we only drop one feature at a time, but it suggests that it is generic. I'd suggest making this function return _all_ dropped feature names, even if there are multiple, so that it stays generic. And then we can require single-feature-drop in `validateFeatureRemovalAtSnapshot`. That function is specific to when the current client drops features, so it's OK to do something specific there.